dev/fattest.simplicity/src/componenttest/rules/repeater/FeatureReplacementAction.java

```diff
@@ -401,9 +432,126 @@ public boolean isEnabled() {
     public void setup() throws Exception {
         final String m = "setup";
 
-        //check that there are actually some features to be added or removed
-        assertFalse("No features were set to be added or removed", addFeatures.size() == 0 && removeFeatures.size() == 0);
+        //check that there are actually some features or options to be added or removed
+        assertFalse("No features or options were set to be added or removed", addFeatures.size() == 0 && removeFeatures.size() == 0 && optionsToAdd.size() == 0);
+
+        // Manage JVM options
+        if (optionsToAdd.size() != 0)
+            setJvmOptions();
 
+        // Manage feature changes
+        if (addFeatures.size() != 0 || removeFeatures.size() != 0)
+            setFeatures();
+    }
+
+    @Override
+    public void cleanup() {
+        // Undo changes done to jvm.options
+        for (Entry<File, File> mapping : optionsFileBackupMapping.entrySet()) {
+            File original = mapping.getKey();
+            File backup = mapping.getValue();
+            Log.info(c, "cleanup", "Restoring " + original + " from " + backup);
+            try {
+                FileUtils.recursiveDelete(original);
+                FileUtils.copyDirectory(backup, original);
+            } catch (Exception e) {
+                throw new RuntimeException("Exception restoring backup for file " + original, e);
+            }
+        }
+        optionsFileBackupMapping.clear();
```

Comment:
Maybe this all should be in an  `if(!optionsFileBackupMapping.isEmpty())`?  Also do you need to reset the LibertyServer and LibertyClient like you did in the setJvmOptions?

Not sure if clear is needed because it is local to just this instance of FeatureReplacementAction.  It isn't static, but it doesn't hurt for it to be here either.

Should backup be deleted after so it is like a move instead of a copy when restoring from backup?

Reply:
Makes sense to wrap it under the `isEmpty` check so I'll do that. I'm not sure if it's necessary since the servers get reset on every setup of the RepeatActions we have if I'm not mistaken. Thought it could help save some time between repeats to not reset the servers, but if you think it can have another effect I haven't thought about I can reset them on cleanup too.

Yeah I agree the clear isn't necessary here just wanted to add it as clean up.

Yeah I think it would be nice to delete the backup folder on cleanup to save up memory space. I'll make the change.